pkg/api/dashboard.go

```diff
@@ -230,6 +230,13 @@ func deleteDashboard(c *models.ReqContext) response.Response {
 		return response.Error(500, "Failed to delete dashboard", err)
 	}
 
+	if hs.Cfg.IsPanelLibraryEnabled() {
```

Comment:
Doesn't it make more sense to disconnect library panels from the dashboard _before_ deleting the dashboard? If the database had foreign key constraints, deleting the dashboard first could fail due to the references. Also, references could potentially be left dangling after deleting the dashboard, in case the following deletion doesn't work.

Reply:
Ok, that makes sense, just remember there are no foreign keys in Grafana :)

Reply:
@hugohaggmark Yeah I am aware, but it's good to think as if there could be. The assumption could break at some point.

Reply:
Fixed!